mhkit/wave/io/wave_hindcast.py

```diff
@@ -0,0 +1,84 @@
+import pandas as pd
+import numpy as np
+from rex import WaveX, MultiYearWaveX
+
+def read_US_wave_dataset(wave_path, parameter, lat_lon, years=None, tree=None, 
+                                 unscale=True, str_decode=True, hsds=True):
+    
+        """
+        Reads data from the WPTO wave hindcast data hosted on AWS. 
+
+        Note: To access the WPTO hindcast data, you will need to configure h5pyd for data access on HSDS. 
+        To get your own API key, visit https://developer.nrel.gov/signup/. 
+
+        To configure h5phd type 
+        hsconfigure
+        and enter at the prompt:
+        hs_endpoint = https://developer.nrel.gov/api/hsds
+        hs_username = None
+        hs_password = None
+        hs_api_key = {your key}
+        Parameters
+        ----------
+        wave_path : string
+            Path to US_Wave .h5 files
+            Available formats:
+                /nrel/US_wave/US_wave$_{year}.h5
+                /nrel/US_wave/US_wave$_*.h5 (Only use when specifying years parameter)
+        parameter: string
+            dataset parameter to be downloaded
+            spatial dataset options: directionality_coefficient, energy_period, maximum_energy_direction
+                mean_absolute_period, mean_zero-crossing_period, omni-directional_wave_power, peak_period
+                significant_wave_height, spectral_width, water_depth 
+        lat_lon: tuple or list of tuples
+            latitude longitude pairs at which to extract data 
+        years : list, optional
+            List of years to access, by default None
```

Comment:
What is the range of years available? If the default is `None` does that mean that all years are returned by default?

Reply:
No, all years are not returned by default unless you use the * wildcard in the data path and leave years as None. Good idea to add the available years to the Docstring

Reply:
addressed in https://github.com/MHKiT-Software/MHKiT-Python/pull/60/commits/151327767fd268ae8b9cae9beda9f68b84cd23b1